src/compiler/scala/tools/nsc/Reporting.scala

```diff
@@ -774,7 +774,8 @@ object Reporting {
       private[this] val cache = mutable.Map.empty[SourceFile, Boolean]
 
       def check(pos: Position) = cache.getOrElseUpdate(pos.source, {
-        val sourcePath = pos.source.file.canonicalPath.replace("\\", "/")
+        val sourceFile = pos.source.file.absolute.file
+        val sourcePath = sourceFile.toPath.normalize.toString.replace("\\", "/")
```

Comment:
`source.file` is an `AbstractFile`, which has a bunch of subtypes. We need to make sure what we do here works / is safe for all of them; in particular, `VirtualFile.file` seems to be `null`.

Sad that this didn't turn up in our test suite...

```suggestion
        val sourcePath = Option(sourceFile).map(_.toPath.normalize.toString).getOrElse(pos.source.path).replace("\\", "/")
```

Reply:
Done in ce5c0c68cfd7dca2e7cd838cbd5e95fb3ff8d515. Added `WConfTest.sourcePatternMatchesVirtualFiles` to validate the before and after behavior.